lib/view_component/slot_v2.rb

```diff
@@ -27,9 +31,15 @@ def to_s
       view_context = @parent.send(:view_context)
 
       @content = if defined?(@_component_instance)
-        # render_in is faster than `parent.render`
-        if defined?(@_content_block)
+        if defined?(@_content_set_by_with_content)
```

Comment:
Will we raise if there's an `@_content_block` and an `@_content_set_by_with_content` value passed?

Reply:
hmm we could to keep it more consistent, do you think it is worth it?

Reply:
If someone tries to pass a block to a slot that has already had `with_content` called on it I think we should raise instead of silently overriding one, since it could save some amount of debugging time.

Reply:
added in https://github.com/github/view_component/pull/875/commits/87db42624b8eaf5b539ec5d26650af9e8567c526